EventFilter/SiPixelRawToDigi/test/test_GPUSimpleVector.cu

```diff
@@ -0,0 +1,63 @@
+//  author: Felice Pantaleo, CERN, 2018
+#include "EventFilter/SiPixelRawToDigi/interface/GPUSimpleVector.hpp"
+#include <cassert>
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include <iostream>
+#include <new>
+
+__global__ void atomic_vector(GPU::SimpleVector<int> *foo) {
+  auto index = threadIdx.x + blockIdx.x * blockDim.x;
+  foo->push_back_ts(index);
+}
+
+int main() {
+  auto maxN = 10000;
+  GPU::SimpleVector<int> *obj_ptr = nullptr;
+  GPU::SimpleVector<int> *d_obj_ptr = nullptr;
+  GPU::SimpleVector<int> *tmp_obj_ptr = nullptr;
+  int *data_ptr = nullptr;
+  int *d_data_ptr = nullptr;
+
+  bool success =
+      cudaMallocHost(&obj_ptr, sizeof(GPU::SimpleVector<int>)) == cudaSuccess &&
+      cudaMallocHost(&data_ptr, maxN * sizeof(int)) == cudaSuccess &&
+      cudaMalloc(&d_data_ptr, maxN * sizeof(int)) == cudaSuccess;
```

Comment:
How about eventually moving the memory management as part of the vector as well? (maybe it would not be "simple" anymore) Now one has to manage both the vector object (=two ints and a pointer) and the memory for the data.